lib/phoenix_component.ex

```diff
@@ -2868,4 +2868,49 @@ defmodule Phoenix.Component do
     %><% end %>
     """
   end
+
+  @doc """
+  Renders an async assign with slots for the different loading states.
+
+  ## Examples
+
+  ```heex
+  <.async_result :let={org} assign={@org}>
+    <:loading>Loading organization...</:loading>
+    <:empty>You don't have an organization yet</:error>
+    <:error :let={{_kind, _reason}}>there was an error loading the organization</:error>
+    <%= org.name %>
+  <.async_result>
+  ```
+  """
+  attr.(:assign, :any, required: true)
+  slot.(:loading, doc: "rendered while the assign is loading")
+
+  # TODO decide if we want an empty slot
+  slot.(:empty,
+    doc:
+      "rendered when the result is loaded and is either nil or an empty list. Receives the result as a :let."
+  )
+
+  slot.(:failed,
```

Comment:
the docs above use `:error` but the code is using `render_slot(@failed, ...`

which one is correct?

Reply:
```suggestion
  slot.(:error,
```